torch/csrc/jit/tracer.h

```diff
@@ -233,4 +253,24 @@ inline bool VariableFlags::verify(const Variable& var) {
   return !was_null && requires_grad == var.requires_grad() && is_volatile == var.is_volatile();
 }
 
+Node* recordTraceWorker(std::string op, at::ArrayRef<Variable> inputs, at::ArrayRef<Variable> outputs);
+
+inline Node* recordTrace(std::string op, std::initializer_list<Variable> inputs, const Variable& output) {
+  return recordTraceWorker(op, inputs, {output});
+}
+inline Node* recordTrace(std::string op, std::initializer_list<Variable> inputs, const std::tuple<Variable, Variable>& outputs) {
+  return recordTraceWorker(op, inputs, {std::get<0>(outputs), std::get<1>(outputs)});
+}
+inline Node* recordTrace(std::string op, std::initializer_list<Variable> inputs, const std::tuple<Variable, Variable, Variable>& outputs) {
+  return recordTraceWorker(op, inputs, {std::get<0>(outputs), std::get<1>(outputs), std::get<2>(outputs)});
+}
+inline Node* recordTrace(std::string op, std::initializer_list<Variable> inputs, at::TensorList& outputs) {
+  // TODO: Eliminate the intermediate vector allocation
+  return recordTraceWorker(op, inputs, std::vector<Variable>(outputs.begin(), outputs.end()));
+}
+inline Node* recordTrace(std::string op, at::TensorList inputs, const Variable& output) {
```

Comment:
Uhhhh all these overloads are really ugly

Reply:
BTW when are inputs an `at::TensorList`? It seems to me that you're always wrapping them in an initializer list

Reply:
TensorList input arguments occur for `cat`. I am always wrapping in an initializer list, but ISTR that if I say `std::initializer_list<at::TensorList>` the overload is ambiguous. The overload this way is consistent with how the `_flags` method works.

Reply:
The overloads are intended to simplify code generation; it's not to avoid allocations. The need for a pile of outputs overloads is due to the current codegen strategy.